packages/cli/src/site-configs.types.ts

```diff
@@ -0,0 +1,22 @@
+export type SiteConfig<ContentScope = object, Private = object, Public = object> = {
+    name: string;
+    contentScope: ContentScope;
+    domains: {
+        main: string;
+        preliminary?: string;
+        pattern?: string;
+        cdn?: string;
+        additional?: string[];
+        preview: string;
+    };
+    preloginEnabled?: boolean;
+    public?: Public;
+} & Private;
+export type SiteConfigPrivate<S extends SiteConfig<object, object, object>> = Omit<S, "public"> & {
+    url: string;
+    previewUrl: string;
+} & S["public"];
```

Comment:
I'm not sure if it is a good idea to move the public object out: with that you have 2 SiteConfigs (definition vs usage) and one has nested public and the other not.